File: lib/src/main/java/com/sd/lib/http/body/FileBody.java
package com.sd.lib.http.body;

import android.text.TextUtils;

import java.io.File;
import java.net.HttpURLConnection;

public class FileBody extends BaseBody<File>
{
    private final File mFile;

    public FileBody(File file)
    {
        if (file == null)
            throw new NullPointerException("file is null");
        mFile = file;
    }

    @Override
    public final String getContentType()
    {
        final String contentType = HttpURLConnection.guessContentTypeFromName(mFile.getName());
        if (!TextUtils.isEmpty(contentType))
            return contentType;

        return super.getContentType();
    }

    @Override
    public final File getBody()
    {
        return mFile;
    }
}

File: lib/src/main/java/com/sd/lib/http/body/FileBody.kt
package com.sd.lib.http.body

import com.sd.lib.http.ContentType
import java.io.File
import java.net.HttpURLConnection

class FileBody(file: File) : IRequestBody<File> {

    override val body: File = file

    override val contentType: String by lazy {
        var contentType = HttpURLConnection.guessContentTypeFromName(body.name)
        if (contentType == null || contentType.isEmpty()) {
            contentType = ContentType.STREAM
        }
        contentType
    }
}